src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxFactoryTests.cs

```diff
@@ -119,6 +120,31 @@ public void TestFreeFormTokenFactory_SpecialTokenKinds()
         }
 
         [Fact]
+        public void TestFreeFormTokenFactoryExceptionMessage_SpecialTokenKinds()
+        {
+            // Make sure the exception message displays the correct namespace/class to use
+            List<SyntaxKind> syntaxKinds = new List<SyntaxKind>()
+            {
+                SyntaxKind.IdentifierToken,
+                SyntaxKind.CharacterLiteralToken,
+                SyntaxKind.NumericLiteralToken
+            };
+            foreach (var kind in syntaxKinds)
+            {
+                try
+                {
+                    SyntaxFactory.Token(default(SyntaxTriviaList), kind, "text", "valueText", default(SyntaxTriviaList));
+                    AssertEx.Fail("Should have thrown");
+                    return;
+                }
+                catch (ArgumentException e)
+                {
+                    Assert.Contains("Microsoft.CodeAnalysis.CSharp.SyntaxFactory", e.Message);
```

Comment:
I think we should verify the entire message. Make sure to use ``using (new EnsureEnglishUICulture())``` around the try/catch. Also, I would expect the test to verify that the APIs suggested by the messages exist and work as suggested.